### PR TITLE
Update data files to CLDR v. 29 beta and related cleanup

### LINKED ISSUE #15: Implement time zone city names
CLDR has complete and updated data on time zone names, in `{CLDR_locale}/timeZoneNames.json`. See http://cldr.unicode.org/translation/timezones for some high level background.

### LINKED ISSUE #54: update unit tests to check for presence of JSON files for all wp locales
let's add unit tests to confirm we have all needed files. at the moment, for example, `get_cldr_locale()` for the .org locale `als` returns `gsw` which is a valid CLDR language code but is one of the locales like `pt-br` which have a different language code than the locale they use for JSON files/trees ( 😡).

starting point that could be added to `test_all_WordPress_locales()`:
```
		foreach ( $wp_locales as $wp_locale ) {
			$cldr_locale = $this->cldr->get_cldr_locale( $wp_locale );
			$this->assertNotEmpty( $cldr_locale );
			$cldr_json_file = dirname( __DIR__ ) . "/json/28.0.2/cldr-localenames-modern/main/$cldr_locale/territories.json";
			$this->assertFileExists( $cldr_json_file );
		}
```
but we don't want to have to update version number so probably call `get_cldr_json_file()` instead.

### PR DESCRIPTION
**Pull request by stuwest**
This PR implements CLDR29 (in beta now and out officially Wednesday March 16) and makes related underlying changes.

First commit updates the scripts for importing CLDR data to:
- improve the mapping of WordPress locales to CLDR locales; update locale #unit tests which resolves #54.
- include more CLDR JSON data items/files with time zone information and date/time formatting which are necessary to address #24, #41, #51.
- exclude unnecessary files resolving #53.
- reorganize the way files are stored which will make it easier to address #37.
- make related changes to class and readmes.

Second commit is the 1200+ data files changes to add CLDR v29 data (via the first commit's new scripts) and remove CLDR v28 data.

Third commit demonstrates potential for new data with methods that return time zone city names, related tests, and demo on settings page. This resolves #15 and creates foundation to address #50.

### CODE REVIEW

**Comment by stuwest** [comment]
I did some squashing to clean up the commits and separate out all the data file changes into their own commit.

**Comment by jblz** [comment]
Nice :) I didn't review the data files commit (043406a), but most everything else looks good. I left a few line comments.

Please go ahead and merge the data files (or wait until the refresh comes out on Wednesday). It'll make this PR a lot more manageable.

**Comment by stuwest** [comment]
OK i added another commit with updates.

I don't want to wait for the final CLDR 29 to get this into the plugin review queue (that Wednesday release date might slip). So I merged the new files into master and did a rebase of this branch dropping the commit that included the files. Then I did a `git push -f`. I thought that was the right idea but not so much since it looks like this PR lost your super helpful line comments and my erudite replies. Git-fail. I had the PR open in another tab so captured the below PDF with our back-and-forth.

LMK how it looks. Is this a case where I should squash all the PR commits into a single one before merging?

[Update data files to CLDR v. 29 beta and related cleanup by stuwest · Pull Request #80 · Automattic:wp-cldr.pdf](https://github.com/Automattic/wp-cldr/files/173360/Update.data.files.to.CLDR.v.29.beta.and.related.cleanup.by.stuwest.Pull.Request.80.Automattic.wp-cldr.pdf)

File: wp-cldr.php [hunk 1]
```diff
@@ -62,8 +62,10 @@ function wp_cldr_settings() {
 
 	<div class="wrap">
 	<h1><?php esc_html_e( 'WP CLDR settings and info', 'wp-cldr' ); ?></h1>
-	<?php esc_html_e( 'This plugin gives WordPress developers easy access to localized country, region, language, currency, and calendar info from the Unicode Common Locale Data Repository. Here are some examples:', 'wp-cldr' );
-	echo '<br>'; ?>
+	<?php esc_html_e( 'This plugin gives WordPress developers easy access to localized country, region, language, currency, and calendar info from the Unicode Common Locale Data Repository. See examples below. More at: ' ); ?>
```
Comment (jblz) [inline wp-cldr.php]: 'wp-cldr' text domain is missing in this call.

**Comment by jblz** [comment]
This looks good to merge (once that text domain is put in).

I have a couple of translation-related suggestions, but I'll make a separate PR for simplicity sake.
Great job!

**Comment by stuwest** [comment]
Squashed commits into 1. Lost the line comments again argh. There must be a way to preserve them.

**Pull request merged**